src/client/datascience/editor-integration/codeLensFactory.ts

```diff
@@ -65,7 +65,7 @@ export class CodeLensFactory implements ICodeLensFactory {
         this.configService.getSettings(undefined).onDidChange(this.onChangedSettings, this, disposables);
         notebook.onDidChangeNotebookCellExecutionState(this.onDidChangeNotebookCellExecutionState, this, disposables);
         kernelProvider.onDidDisposeKernel(
-            (kernel) => this.notebookData.delete(kernel.notebookUri.toString()),
+            (kernel) => this.notebookData.delete(kernel.notebookDocument.uri.toString()),
```

Comment:
Added NotebookDocument as a property on the kernel we don't need the Uri, we can pass the Document directly (we didn't pass Document as it wasn't possible in interacive, now it is)